src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -453,6 +462,50 @@ class _BuildPexRequirementsSetup:
     concurrency_available: int
 
 
+@dataclass(frozen=True)
+class ReqStrings:
+    req_strings: tuple[str, ...]
+
+
+@rule
+async def get_req_strings(pex_reqs: PexRequirements) -> ReqStrings:
+    addrs: list[Address] = []
+    specs: list[str] = []
+    req_strings: list[str] = []
+    for req_str_or_addr in pex_reqs.req_strings_or_addrs:
+        if isinstance(req_str_or_addr, Address):
+            addrs.append(req_str_or_addr)
+        else:
+            assert isinstance(req_str_or_addr, str)
+            # A local or VCS requirement will have an ampersand in it, and any other
```

Comment:
Reviewers, do you concur?

Reply:
I think so. But you could also explicitly recommend prefixing an address with `//` in case of ambiguity, and skip the `@` check (note: not ampersand).

Reply:
Whoops, I meant asperand.

Reply:
We would have to require prefixing, we can't just recommend it. Otherwise we'll see vcs/local requirements as addresses with no escape hatch.

Reply:
And yeah, I think we should require the `//` prefix, because apparently wheel paths can be treated as requirements.